tests/try_scraper/test_try_scraper.py

```diff
@@ -6,73 +6,55 @@
-     'url': 'firefox/try-builds/test-user@mozilla.com-8fcac92cfcad/'
-            'try-linux-debug/firefox-38.0a1.en-US.linux-i686.tar.bz2'},
+    ({'platform': 'linux', 'revision': '8fcac92cfcad', 'debug_build': True},
+     '8fcac92cfcad-debug-firefox-38.0a1.en-US.linux-i686.tar.bz2',
+     'firefox/try-builds/test-user@mozilla.com-8fcac92cfcad/'
+     'try-linux-debug/firefox-38.0a1.en-US.linux-i686.tar.bz2'),
     # -a firefox -p win32 --revision=8fcac92cfcad
-    {'args': {'platform': 'win32',
-              'revision': '8fcac92cfcad'},
-     'filename': '8fcac92cfcad-firefox-38.0a1.en-US.win32.installer.exe',
-     'url': 'firefox/try-builds/test-user@mozilla.com-8fcac92cfcad/'
-            'try-win32/firefox-38.0a1.en-US.win32.installer.exe'},
+    ({'platform': 'win32', 'revision': '8fcac92cfcad'},
+     '8fcac92cfcad-firefox-38.0a1.en-US.win32.installer.exe',
+     'firefox/try-builds/test-user@mozilla.com-8fcac92cfcad/'
+     'try-win32/firefox-38.0a1.en-US.win32.installer.exe'),
     # -a firefox -p win64 --revision=8fcac92cfcad
-    {'args': {'platform': 'win64',
-              'revision': '8fcac92cfcad'},
-     'filename': '8fcac92cfcad-firefox-38.0a1.en-US.win64.installer.exe',
-     'url': 'firefox/try-builds/test-user@mozilla.com-8fcac92cfcad/'
-            'try-win64/firefox-38.0a1.en-US.win64.installer.exe'},
-]
-
-tests = firefox_tests
-
-
-class TryScraperTest(mhttpd.MozHttpdBaseTest):
-    """test mozdownload TryScraper class"""
-
-    @patch('mozdownload.treeherder.Treeherder.query_builds_by_revision')
-    def test_scraper(self, query_builds_by_revision):
-        """Testing various download scenarios for TryScraper"""
-        query_builds_by_revision.return_value = [
-            '/firefox/try-builds/test-user@mozilla.com-8fcac92cfcad/try-foobar/'
-        ]
-
-        for entry in tests:
-
-            scraper = TryScraper(destination=self.temp_dir,
-                                 base_url=self.wdir,
-                                 logger=self.logger,
-                                 **entry['args'])
-            expected_filename = os.path.join(self.temp_dir, entry['filename'])
-            self.assertEqual(scraper.filename, expected_filename)
-            self.assertEqual(urllib.unquote(scraper.url),
-                             urljoin(self.wdir, entry['url']))
+    ({'platform': 'win64', 'revision': '8fcac92cfcad'},
+     '8fcac92cfcad-firefox-38.0a1.en-US.win64.installer.exe',
+     'firefox/try-builds/test-user@mozilla.com-8fcac92cfcad/'
+     'try-win64/firefox-38.0a1.en-US.win64.installer.exe')
+])
+
+
+@patch('mozdownload.treeherder.Treeherder.query_builds_by_revision')
+def test_scraper(mock_query_builds_by_revision, httpd, tmpdir, args, filename, url):
+    """Testing various download scenarios for TryScraper"""
+    mock_query_builds_by_revision.return_value = [
+        '/firefox/try-builds/test-user@mozilla.com-8fcac92cfcad/try-foobar/'
+    ]
+    baseurl = httpd.get_url()
```

Comment:
The getter for the url is cheap, so simply get rid of this extra variable declaration/usage.